src/app/client/src/app/modules/learn/services/course-consumption/course-consumption.service.ts

```diff
@@ -19,6 +19,7 @@ export class CourseConsumptionService {
   launchPlayer = new EventEmitter<any>();
   updateContentState = new EventEmitter<any>();
   showJoinCourseModal = new EventEmitter<any>();
+  disableJoinCourseBtn = new EventEmitter();
```

Comment:
instead of  **disableJoinCourseBtn** it should be **enableCourseEntrollment** and emit for both  when openBatchCount is 0 as **false** if more than one then as **true**. use this directly in html template with help of async pipe